File: Python/Algorithms/Divide-Two-Integers.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        q = int(dividend/divisor)
        if q > pow(2, 31)-1:
            return pow(2, 31)-1
        elif q < -pow(2, 31):
            return -pow(2, 31)
        else:
            return q

File: Golang/Divide-Two-Integers.go
func divide(dividend int, divisor int) int {
    
    q := int(dividend/divisor)
    
    maxInt := int(math.Pow(2, 31)-1)
    minInt := int(math.Pow(2, 31)/-1)
    
    if q > maxInt {
        return maxInt
    } else if q < minInt {
        return minInt
    } else {    
        return q
    }
}
